soroban-env-host/src/host.rs

```diff
@@ -842,11 +840,18 @@ impl Host {
         args: &[RawVal],
     ) -> Result<RawVal, HostError> {
         // Create key for storage
-        let storage_key = self.contract_source_ledger_key(id.metered_clone(&self.0.budget)?);
-        match self.retrieve_contract_source_from_storage(&storage_key)? {
+        let storage_key = self.contract_source_ledger_key(id)?;
+        match self.retrieve_contract_source_from_storage(storage_key)? {
             #[cfg(feature = "vm")]
             ScContractCode::WasmRef(wasm_hash) => {
-                let code_entry = self.retrieve_contract_code_from_storage(wasm_hash)?;
+                let key = self.contract_code_ledger_key(&wasm_hash)?;
```

Comment:
I initially removed this helper because it was a little hard to refactor it with the change of interface to `Rc` (borrow checker related). But I realize this is probably not a good reason for removing it.
With the `Rc->&Rc` fix up, this makes it a lot easier to work with. 
So I have added it back.